kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -91,10 +91,26 @@ class SparkProcessBuilder(
       buffer += s"${convertConfigKey(k)}=$v"
     }
 
-    // iff the keytab is specified, PROXY_USER is not supported
-    if (!useKeytab()) {
-      buffer += PROXY_USER
-      buffer += proxyUser
+    // For spark on kubernetes, spark pod using env SPARK_USER_NAME as current user
+    def setSparkUserName(userName: String): Unit = {
+      clusterManager().foreach(cm => {
+        if (cm.startsWith("k8s://")) {
```

Comment:
there is no harm to set for all cluster managers？

Reply:
`spark.kubernetes.driverEnv.` only affect on kubernetes case. We are only setting up this On Kubernetes for now.
As for `SPARK_USER_NAME` this env, it's affect in `spark/kubernetes/entrypoint.sh`